addons/mass_mailing/models/mailing.py

```diff
@@ -293,13 +296,33 @@ def _compute_reply_to(self):
             elif mailing.reply_to_mode == 'update':
                 mailing.reply_to = False
 
-    @api.depends('mailing_model_name', 'contact_list_ids')
+    @api.depends('mailing_model_name', 'contact_list_ids', 'mailing_favorite_filter_id')
     def _compute_mailing_domain(self):
         for mailing in self:
             if not mailing.mailing_model_name:
                 mailing.mailing_domain = ''
             else:
-                mailing.mailing_domain = repr(mailing._get_default_mailing_domain())
+                default_filter = repr(mailing._get_default_mailing_domain())
+                if mailing.mailing_favorite_filter_id:
+                    # If model is changed and filter is set, domain should reset to default one, emptying the filter,
+                    # or, if filter itself is changed (means model is same), simply apply filter's domain
+                    mailing.mailing_domain = mailing.mailing_favorite_filter_id.mailing_domain if mailing.mailing_model_name == mailing.mailing_favorite_filter_id.mailing_model_name else default_filter
+                else:
```

Comment:
Indeed its hackish. But we want to keep the domain of the last deleted filter. That means, user could change n number of filters, but we need to keep the domain from last filter, not the saved one we get from _origin.

Reply:
This is why we only reset the domain to "default_filter" if there is no filter_id AND if it was not just removed.
I was not suggesting to restore the one from _origin here, just to check if it's there to avoid resetting the filter.
Did you try it?

Reply:
hello @awa-odoo 
i will try this,  but the domain is changed or different from last one 
and _origin gives last saved record in db and it returns None if 
record is first time created.

Reply:
In that case can you maybe do this:

```py
elif mailing.mailing_model_name != mailing._origin.mailing_model_name:
    # changed selected model, reset to default filter
    mailing.mailing_domain = default_filter
```

Make sure it works for every use cases of course.

I would really like to avoid that manual filter change in the JS.

Reply:
Hello @awa-odoo 
So did I, but could not find a proper way to deal with it in python :) Let me rephrase a bit again, to explain what we want. If you are creating a new record, there will be nothing in `_origin`. You could change the model as many times as you want, and you could create as many filters for any model as you want. But when you simply click on `star` icon to remove the last created filter, only the filter needs to be deleted but we need to keep the domain(note that we haven't even saved the record). How do we manage that with compute method? I think that all the conditions will lead to default filter being set, because there's no point of comparing anything with `_origin` state for new records. And it will be the case most of times in context of this task(people will play with domains / filters on new records).

Reply:
Yes I see what you mean.

I guess we can't help it then, it's not possible to do only in python.
Unless maybe we start playing with onchange but those are a bit deprecated anyway.

Thanks for trying and for keeping up with me!